.devcontainer/devcontainer.json

```diff
@@ -0,0 +1,53 @@
+{
+	"name": "Eraser",
+	"image": "ghcr.io/cse-labs/k3d:latest",
+
+	// do not change these if you want Docker in Docker support
+	"runArgs": ["--init", "--privileged"],
+	"mounts": [ "source=/var/run/docker.sock,target=/var/run/docker-host.sock,type=bind" ],
+	"overrideCommand": false,
+
+	// some base images require a specific user name
+	"remoteUser": "vscode",
+
+	// Set container specific settings
+	"settings": {
+		"files.trimTrailingWhitespace": true,
+		"files.trimFinalNewlines": true,
+		"files.insertFinalNewline": true
+	},
+
+	"hostRequirements": {
+		"cpus": 4
+	},
+
+	// forward ports for the app
+	"forwardPorts": [
+	],
+
+	// add labels
+	"portsAttributes": {
+	},
+
+	// Install extensions
+	"extensions": [
+	 	"ms-azuretools.vscode-docker",
+	 	"ms-kubernetes-tools.vscode-kubernetes-tools",
+	 	"davidanson.vscode-markdownlint",
+	 	"gruntfuggly.todo-tree",
+	 	"humao.rest-client",
+		"ms-vscode.makefile-tools",
+        "golang.go"
+	],
+
+	"waitFor": "postCreateCommand",
+
+	// Use 'onCreateCommand' to run commands as part of container creation.
+	// "onCreateCommand": "/bin/bash -c .devcontainer/on-create.sh",
+
+	// Use 'postCreateCommand' to run commands after the container is created.
+	"postCreateCommand": "/bin/bash -c .devcontainer/post-create.sh"
+
+	// Use 'postStartCommand' to run commands after the container starts.
+	// "postStartCommand": "/bin/bash -c .devcontainer/post-start.sh"
```

Comment:
remove comments?

Reply:
I am more than happy to remove these. To be honest, these are best practices we've found using codespaces, and pretty difficult to find in the documentation. I kept these comments for ease of development. Would you still like me to remove these?